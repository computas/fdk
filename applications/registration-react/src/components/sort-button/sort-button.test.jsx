import React from 'react';
import { shallow } from 'enzyme';
import SortButtons from './sort-button.component';

let defaultProps;
let wrapper;
let onSortField;

beforeEach(() => {
  onSortField = jest.fn();
  defaultProps = {
    field: 'sortField',
    sortField: 'sortField',
    sortType: 'asc',
    onSortField
  };
  wrapper = shallow(<SortButtons {...defaultProps} />);
});

test('should render SortButtons correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSortField asc', () => {
  wrapper
    .find('button')
    .first()
    .prop('onClick')();
  expect(onSortField).toHaveBeenCalled();
});

test('should handle onSortField desc', () => {
  wrapper.setProps({
    sortType: 'desc'
  });
  wrapper
    .find('button')
    .at(1)
    .prop('onClick')();
  expect(onSortField).toHaveBeenCalled();
});
