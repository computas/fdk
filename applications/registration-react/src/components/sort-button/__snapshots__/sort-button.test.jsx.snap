// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SortButtons correctly 1`] = `
<div
  className="d-flex flex-column"
>
  <button
    className="d-flex sortButton transparentButton visibilityHidden"
    name="sortFieldasc"
    onClick={[Function]}
    title="Stigende"
  >
    <i
      className="fa fa-sort-up fdk-color0"
    />
  </button>
  <button
    className="d-flex sortButton transparentButton "
    name="sortFielddesc"
    onClick={[Function]}
    title="Synkende"
  >
    <i
      className="fa fa-sort-down fdk-color0"
    />
  </button>
</div>
`;
